src/scenes/Home/scenes/ManageMentees/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from 'react';
+import { Typography, List, notification, Spin, Empty, Row, Col } from 'antd';
+import { Mentee, SavedProgram } from '../../../../interfaces';
+import { useParams } from 'react-router';
+import axios, { AxiosResponse } from 'axios';
+import styles from '../../styles.css';
+import MenteeRow from './components/MenteeRow';
+
+const { Title } = Typography;
+
+function ManageMentees() {
+  const { mentorId, programId } = useParams();
+  const [mentees, setMentees] = useState<Mentee[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [program, setProgram] = useState<SavedProgram | null>(null);
+  const [shouldLoadMentees, setShouldLoadMentees] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getProgram();
+  }, []);
+
+  const getProgram = () => {
+    axios
+      .get(`http://localhost:8080/api/scholarx/programs/${programId}`)
+      .then((result: AxiosResponse<SavedProgram>) => {
+        if (result.status == 200) {
+          setProgram(result.data);
+          const shouldLoadMentees =
+            result.data.state == 'MENTEE_APPLICATION' ||
+            result.data.state == 'MENTEE_SELECTION' ||
+            result.data.state == 'ONGOING';
+          setShouldLoadMentees(shouldLoadMentees);
+          if (shouldLoadMentees) {
+            getMentees();
+          }
+        } else {
+          throw new Error();
+        }
+      })
+      .catch(() => {
+        setIsLoading(false);
+        notification.error({
+          message: 'Error!',
+          description: 'Something went wrong when fetching the program detail',
+        });
+      });
+  };
+
+  const getMentees = () => {
+    axios
+      .get(`http://localhost:8080/mentors/${mentorId}/mentees`)
+      .then((result: AxiosResponse<Mentee[]>) => {
+        if (result.status == 200) {
+          setMentees(result.data);
+          setIsLoading(false);
+        } else {
+          throw new Error();
+        }
+      })
+      .catch(() => {
+        setIsLoading(false);
+        notification.error({
+          message: 'Error!',
+          description: 'Something went wrong when fetching the mentees',
+        });
+      });
+  };
+
+  return (
+    <div className={styles.container}>
+      <Spin tip="Loading..." spinning={isLoading}>
+        <Row>
+          <Col md={3} />
+          <Col md={15}>
+            <Title>Manage Mentees</Title>
+            {!shouldLoadMentees && program != null && (
```

Comment:
```suggestion
            {program && !shouldLoadMentees && (
```
Try this, I think it should work

Reply:
In this case, we know the possible values for the program variable is only `null` OR a `SavedProgram`

`{program && ...` checks whether the program is

- null
- undefined
- NaN
- empty string ("")
- 0
- false

Reply:
Is it going to check all of them? because we already initiated `program` as null or saved program why would it check for other types?

Reply:
It is going to check all of them according to the JS. (Typescript isn't available at runtime )

Reply:
So in the compiling stage `Program` is can be null or SavedProgram but when it's in runtime it can be anything right because it's Js?

Reply:
No, it can't be, because according to our logic we know the possible values are only null or an object (saved program) Not because we initialized them as that way, but we know all the possible things can be happened according to the written code.  

But, the browser (DOM) does not know that. From this piece of code `{program && ...` we instruct the DOM to check whether the program variable does not equal one of the following,
- null
- undefined
- NaN
- empty string ("")
- 0
- false 

It has to check one by one. 

---
Let's assume the `program`'s value is a program object,

Now, it checks whether it is null. the answer is false.
Then it checks whether it is undefined.  the answer is false.
Then, checks whether it is NaN
then, empty string 
then 0,
then false, 
it is not equal to anything. finally, it say's "hey program, you are good to go!"

Reply:
ah okay Thanks for the explanation!

Reply:
Actually, I used the above example only to explain the situation. The actual thing that happens is by the abstract operation `ToBoolean`

For boolean values `ToBoolean` doesn't do any conversion

Reply:
ok cool

Reply:
:)